okhttp/src/main/java/io/grpc/transport/okhttp/OkHttpClientStream.java

```diff
@@ -96,6 +96,10 @@ public void transportHeadersReceived(List<Header> headers, boolean endOfStream)
     synchronized (lock) {
       if (endOfStream) {
         transportTrailersReceived(Utils.convertTrailers(headers));
+        if (getTransportError() != null) {
```

Comment:
I didn't get you, removing what endOfStream condition? you mean the one in Http2ClientStream.transportDataReceived? but in the normal case the endOfStream come with the trailer, so we also need to call sendCancel() in transportTrailersReceived, is this you want?